SINGER/SINGER/Rate_map.cpp

```diff
@@ -38,10 +39,11 @@ int Rate_map::find_index(double x) {
 }
 
 double Rate_map::cumulative_distance(double x) {
-    int index = find_index(x);
+    const double actualCoordinate = x + offset;
+    int index = find_index(actualCoordinate);
     double prev_dist = rate_distances[index];
     double next_dist = rate_distances[index+1];
-    double p = (x - coordinates[index])/(coordinates[index+1] - coordinates[index]);
+    double p = (actualCoordinate - coordinates[index])/(coordinates[index+1] - coordinates[index]);
     double dist = (1-p)*prev_dist + p*next_dist;
     return dist;
 }
```

Comment:
likely `Rate_map::mean_rate()` also needs to be adjusted based on the region between `start_pos` and `end_pos` instead of taking the mean of the entire map.  hopefully it is not a big deal, given that `parallel_singer` uses regions of `1Mbp` that hopefully have means close to the full mean.

Reply:
This fix makes sense (to me) because the alternative is that user has to adjust the recombination map many times, which could be cumbersome.